symengine/catch/CMakeLists.txt

```diff
@@ -0,0 +1,14 @@
+project(catch)
+
+set(SRC
+    catch.cpp
+)
+
+add_library(catch STATIC ${SRC})
+target_link_libraries(catch symengine)
+
+install(TARGETS catch
+            RUNTIME DESTINATION bin
+            ARCHIVE DESTINATION lib
+            LIBRARY DESTINATION lib
+        )
```

Comment:
So since SymEngine depends on Teuchos headers in Debug mode, this means that user programs also need it, thus Teuchos must be installed. 

SymEngine does not depend on Catch, so it doesn't need to be installed. However, if it is not installed, then the tests can only be compiled by adding an include flag (`-I`) to the `catch.hpp`, which is only located in the source tree. I think that's fine.